docs/sql-migration-guide.md

```diff
@@ -22,6 +22,10 @@ license: |
 * Table of contents
 {:toc}
 
+## Upgrading from Spark SQL 3.2 to 3.3
+
+  - Non-nullable schema was not supported properly in previous Spark version so the output schema of `DataFrameReader.json(jsonDataset: Dataset[String])` and `DataFrameReader.csv(csvDataset: Dataset[String])` became nullable which also matches with `DataFrameReader.json(path: String)` and `DataFrameReader.csv(path: String)`.
```

Comment:
Let's rephrase a bit more and keep the sentence format consistent with others. e.g.) Since Spark 3.3, blah blah .... In Spark 3.1 or earlier, blah blah ...

Reply:
It's a bug fix, as the old versions return the wrong result or fail. Do we need to update the migration guide?

Reply:
Hm, it fails only when the data contains `null`. If the data doesn't contain `null`, it works all good.

Reply:
ok, then shall we backport this fix?

Reply:
hmmm .. maybe let's dont for now just to be conservative ...